Source/Core/VideoBackends/OGL/BoundingBox.cpp

```diff
@@ -13,58 +13,182 @@
 
 static GLuint s_bbox_buffer_id;
 
+static int s_image_mask_bounds[4] = {0, 0, 0, 0};
+static u8* s_image_mask_data = nullptr;
+static GLuint s_image_mask_texture;
+
+static float s_xscale;
+static float s_yscale;
+
 namespace OGL
 {
-void BoundingBox::Init()
+GLuint BoundingBox::image_mask_fbo;
+bool BoundingBox::image_mask_updated;
+bool BoundingBox::image_mask_cleared;
+
+void BoundingBox::SetEfbScaleChanged(float xscale, float yscale)
 {
-  if (g_ActiveConfig.backend_info.bSupportsBBox)
+  s_xscale = xscale;
+  s_yscale = yscale;
+
+  if (!g_ActiveConfig.backend_info.bSupportsFragmentStoresAndAtomics)
+  {
+    image_mask_updated = false;
+
+    if (s_image_mask_data)
```

Comment:
s_image_mask_data can just be a unique_ptr so explicit management like this isn't necessary.